From d9282841496f35002f13e4751ab235e26c60bae5 Mon Sep 17 00:00:00 2001
From: Andreas Dinter <andreas.dinter@systec-electronic.com>
Date: Mon, 5 Aug 2024 14:26:08 +0200
Subject: [PATCH 09/13] board/ti/am62x/sysworxx.c: Import EEPROM as part of
 board_late_init()

This allows to perform board specific actions in u-boot itself and
cleans up `bootcmd` which is most likely not intended for such
operations.

Signed-off-by: Andreas Dinter <andreas.dinter@systec-electronic.com>
---
 board/ti/am62x/sysworxx.c   | 15 ++++++++++++++-
 board/ti/am62x/sysworxx.env |  1 -
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/board/ti/am62x/sysworxx.c b/board/ti/am62x/sysworxx.c
index 11f0ecff086..348949f3680 100644
--- a/board/ti/am62x/sysworxx.c
+++ b/board/ti/am62x/sysworxx.c
@@ -20,6 +20,7 @@
 #include <asm/arch/hardware.h>
 #include <dm/uclass.h>
 #include <asm/gpio.h>
+#include <command.h>
 
 #include "../common/fdt_ops.h"
 #include "../common/k3-ddr-init.h"
@@ -90,7 +91,19 @@ void reset_eth_phys(void) {
 #if CONFIG_IS_ENABLED(BOARD_LATE_INIT)
 int board_late_init(void)
 {
-	reset_eth_phys();
+	printf("EEPROM import\n");
+	int ret = run_command("run eeprom_import", CMD_FLAG_ENV);
+	if (ret == 0) {
+		char* fdt_prefix = env_get("fdt_prefix");
+		char* hw_iface_rev = env_get("hw_iface_rev");
+
+		if ((fdt_prefix != NULL) && (hw_iface_rev != NULL)) {
+			printf("fdt_prefix:   %s\n", fdt_prefix);
+			printf("hw_iface_rev: %s\n", hw_iface_rev);
+		}
+
+		reset_eth_phys();
+	}
 
 	return 0;
 }
diff --git a/board/ti/am62x/sysworxx.env b/board/ti/am62x/sysworxx.env
index d69a1cc03ee..e677c109094 100644
--- a/board/ti/am62x/sysworxx.env
+++ b/board/ti/am62x/sysworxx.env
@@ -7,7 +7,6 @@
 #endif
 
 default_device_tree=ti/k3-am623-systec-fallback.dtb
-bootcmd=run eeprom_import; run envboot; run distro_bootcmd;
 findfdt=
 	setenv name_fdt ${default_device_tree};
 	if test -n "$fdt_prefix" -a -n "$hw_iface_rev"; then
-- 
2.47.0

